Without this patch when -pthreads is useed, _REENTRANT is not set.

diff -wpruN '--exclude=*.orig' a~/include/clang/Driver/Options.td a/include/clang/Driver/Options.td
--- a~/include/clang/Driver/Options.td	1970-01-01 00:00:00
+++ a/include/clang/Driver/Options.td	1970-01-01 00:00:00
@@ -2496,7 +2496,7 @@ def print_resource_dir : Flag<["-", "--"
 def print_search_dirs : Flag<["-", "--"], "print-search-dirs">,
   HelpText<"Print the paths used for finding libraries and programs">;
 def private__bundle : Flag<["-"], "private_bundle">;
-def pthreads : Flag<["-"], "pthreads">;
+def pthreads : Flag<["-"], "pthreads">, Flags<[CC1Option]>;
 def pthread : Flag<["-"], "pthread">, Flags<[CC1Option]>,
   HelpText<"Support POSIX threads in generated code">;
 def no_pthread : Flag<["-"], "no-pthread">, Flags<[CC1Option]>;
diff -wpruN '--exclude=*.orig' a~/lib/Driver/ToolChains/Clang.cpp a/lib/Driver/ToolChains/Clang.cpp
--- a~/lib/Driver/ToolChains/Clang.cpp	1970-01-01 00:00:00
+++ a/lib/Driver/ToolChains/Clang.cpp	1970-01-01 00:00:00
@@ -4539,6 +4539,7 @@ void Clang::ConstructJob(Compilation &C,
                   options::OPT_fno_unroll_loops);
 
   Args.AddLastArg(CmdArgs, options::OPT_pthread);
+  Args.AddLastArg(CmdArgs, options::OPT_pthreads);
 
   if (Args.hasFlag(options::OPT_mspeculative_load_hardening, options::OPT_mno_speculative_load_hardening,
                    false))
diff -wpruN '--exclude=*.orig' a~/lib/Frontend/CompilerInvocation.cpp a/lib/Frontend/CompilerInvocation.cpp
--- a~/lib/Frontend/CompilerInvocation.cpp	1970-01-01 00:00:00
+++ a/lib/Frontend/CompilerInvocation.cpp	1970-01-01 00:00:00
@@ -2479,7 +2479,7 @@ static void ParseLangArgs(LangOptions &O
   if (Args.hasArg(OPT_fzvector))
     Opts.ZVector = 1;
 
-  if (Args.hasArg(OPT_pthread))
+  if (Args.hasArg(OPT_pthread) || Args.hasArg(OPT_pthreads))
     Opts.POSIXThreads = 1;
 
   // The value-visibility mode defaults to "default".
